test/manipulation.test.js

```diff
@@ -293,6 +491,24 @@ describe('manipulation', function () {
         });
       });
     });
+
+    it('should preserve properties with dynamic setters (promise variant)', function(done) {
+      // This test reproduces a problem discovered by LoopBack unit-test
+      // "User.hasPassword() should match a password after it is changed"
+      StubUser.create({ password: 'foo' })
+      .then(function(created) {
+        created.password = 'bar';
+        return created.save()
+        .then(function(saved) {
+          saved.password.should.equal('bar-2');
+          return StubUser.findById(created.id)
+          .then(function(found) {
+            found.password.should.equal('bar-2');
+            done();
+          });
+        });
+      }).catch(done);
```

Comment:
"preserve properties with dynamic setter" is not related to callback vs. promise, can we omit this test?